src/NServiceBus.Testing/TestableMessageHandlerContext.cs

```diff
@@ -0,0 +1,162 @@
+namespace NServiceBus.Testing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Threading.Tasks;
+    using NServiceBus.Extensibility;
+    using NServiceBus.Persistence;
+    using NServiceBus.Testing.ExpectedInvocations;
+
+    internal class TestableMessageHandlerContext : IMessageHandlerContext
```

Comment:
Should this class be `internal`? Are we planning to showcase it as the main test double for the handler context?